packages/store/tests/state.spec.ts

```diff
@@ -84,4 +85,107 @@ describe('Store', () => {
       expect(res).toBe(true);
     });
   });
+
+  describe('given the ngxsOnInit lifecycle method is present', () => {
+    it('should call the ngxsOnInit method on root module initialisation', () => {
+      const listener: string[] = [];
+
+      @State<number>({
+        name: 'foo',
+        defaults: 0
+      })
+      class FooState implements NgxsOnInit {
+        ngxsOnInit(stateContext: StateContext<number>) {
+          listener.push('onInit');
+        }
+      }
+
+      TestBed.configureTestingModule({
+        imports: [NgxsModule.forRoot([FooState])]
+      });
+
+      TestBed.get(FooState);
+
+      expect(listener).toEqual(['onInit']);
+    });
+
+    it('should call the ngxsOnInit method on feature module initialisation', () => {
```

Comment:
can you move these tests to state.spec?

Reply:
Only if it makes sense. Thoughts?

Reply:
Aren't they already in state.spec?

Reply:
Ah yes sorry I thought they were in dispatch